os/arch/xtensa/src/esp32/esp32_i2c.c

```diff
@@ -586,7 +589,7 @@ static int i2c_esp32_transmit(struct i2c_dev_s *dev)
 	}
 #endif
 
-	return 0;
+	return ret;
```

Comment:
Good point here.
please also refer to line 578  
``` 
	if (ret < 0) {
		ret = get_errno();
		return -ETIMEDOUT;
}
```
if TIMEDOUT err returned, no need to get errno?

Reply:
Fixed.